template/logging.tpl

```diff
@@ -1,50 +1,29 @@
-echo 'installing additional software for logging'
-# installing in a loop to ensure the cli is installed.
-for i in {1..7}
-do
-  echo "Attempt: ---- " $i
-  yum install -y aws-cli awslogs jq && break || sleep 60
-done
-
-# Inject the CloudWatch Logs configuration file contents
-cat > /etc/awslogs/awslogs.conf <<- EOF
-[general]
-state_file = /var/lib/awslogs/agent-state
-
-[/var/log/dmesg]
-file = /var/log/dmesg
-log_stream_name = {instanceId}/dmesg
-log_group_name = ${log_group_name}
-initial_position = start_of_file
-
-[/var/log/messages]
-file = /var/log/messages
-log_stream_name = {instanceId}/messages
-log_group_name = ${log_group_name}
-datetime_format = %b %d %H:%M:%S
-initial_position = start_of_file
-
-[/var/log/user-data.log]
-file = /var/log/user-data.log
-log_stream_name = {instanceId}/user-data
-log_group_name = ${log_group_name}
-initial_position = start_of_file
-
+# Create CloudWatch Logs config for GitLab
+cat <<EOF >> /opt/aws/amazon-cloudwatch-agent/etc/amazon-cloudwatch-agent.d/gitlab.json
+{
+  "logs": {
+    "log_stream_name": "{instance_id}",
+    "logs_collected": {
+      "files": {
+        "collect_list": [
+          {
+            "file_path": "/var/log/dmesg",
+            "log_group_name": "{hostname}",
```

Comment:
Change `{hostname}` to `${log_group_name}` to avoid the messiness that appears in the screenshot I attached to the Trello card. After making this change, verify that logs now appear under the `gitlab-runner` log group. After that, please delete all the `ip-...` log groups.

Reply:
@ericamador So in this commit, we are appending to the existing CW log config which is already set up in the base rhel7 image. By default, the base rhel7 image places `/var/log/messages` and `/var/log/secure` logs in the `{hostname}` log group, and we are just adding `dmesg` and `user-data` as well in that same group.

We could change `{hostname}` to `${log_group_name}` here and create the `gitlab-runner` log group, but I don't think it would remove the clutter in CloudWatch (since you still have /var/log/secure and messages logs going there).

Reply:
Great points. After discussing with Alex, this problem is a bit more complex than I initially though, so we're going to address this in a future card.